New Api-version 2025-08-01 for Microsoft.Storage

ujjawaljain-msft opened the pull request:
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![spec_pr_review_workflow_diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/5bb5e7ce-8aff-4dbb-a3f8-0d9b68fef5b1)

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Convert existing [OpenAPI spec to TypeSpec spec](https://aka.ms/typespec/conversion) (do not combine this with implementing changes for a new API version).
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.
- [x] A [release plan](https://aka.ms/azsdkdocs/release-plans) has been created. If not, please create one as it will help guide you through the REST API and SDK creation process. 

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories)
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.
- For guidance on SDK breaking change review, refer to https://aka.ms/ci-fix.

razvanbadea-msft commented on specification/storage/Storage.Management/models.tsp:
```diff
@@ -6048,3 +6167,260 @@ alias ETagHeader = {
+  /**
+   * ARM ResourceId of the managed identity that should be used to authenticate to the backing data source.
+   */
+  identityResourceId?: string;
```
>string [](http://example.com/codeflow?start=23&length=6)

why it is a string and not a managed identity?

razvanbadea-msft commented:
>   #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"

this was added at typespec conversion because you have the context, you should either do it now or create a pr to update the justification or try fixing the suppressions,  there is no reason why this enum does not have a description for its members and adding one it will not do a breaking change. These descriptions are used for automated documentation that will help customers to use your api

---
Refers to: specification/storage/Storage.Management/models.tsp:991 in 5807f17. [](commit_id = 5807f17ce09af4f0979b52f8369c12048b25df34, deletion_comment = False)

ujjawaljain-msft commented:
> > #suppress "@azure-tools/typespec-azure-core/documentation-required" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
> 
> this was added at typespec conversion because you have the context, you should either do it now or create a pr to update the justification or try fixing the suppressions, there is no reason why this enum does not have a description for its members and adding one it will not do a breaking change. These descriptions are used for automated documentation that will help customers to use your api
> 
> Refers to: specification/storage/Storage.Management/models.tsp:991 in 5807f17. [](commit_id = [5807f17](https://github.com/Azure/azure-rest-api-specs/commit/5807f17ce09af4f0979b52f8369c12048b25df34), deletion_comment = False)

I'll need to check why the documentation was not added during typespec conversion. But I agree with you that each member should have a description. Will create a separate PR to add the descriptions and remove the suppressions.

blueww reviewed (changes requested):
When raise the official PR for 2025-08-01, please include all feature changes on this API version (Or after the API version TSP/swagger PR is merged, add new parameters/values might be consider as breaking change).

Per what I know, following are the potential changes on 2025-08-01:

- "allowedCopyScope" and "smart account tier" has mentioned they might will target 2025-08-01 API version before.
- Task Assignment also has a potential new API change: https://github.com/Azure/azure-rest-api-specs/pull/39993/files
- any others?

ujjawaljain-msft commented on specification/storage/Storage.Management/models.tsp:
```diff
@@ -6048,3 +6167,261 @@ alias ETagHeader = {
+   * Consumer visible name of the original path.
+   */
+  displayName: string;
+}
```
I tried adding this locally, but it didn’t result in any changes to the Swagger (.json) file. If you check the Swagger for the first version (2025-06-01), the only difference is the addition of `readOnly: true`, which was done to resolve the LintDiff errors. That change was approved by Mark as well, since it’s effectively a no-op.

The models weren’t added in the first version. I can still apply the versioning decorator to the models if you think that would be useful going forward.

ujjawaljain-msft replied:
I see. Thanks for the context. Updated the PR to include the decorator on newly introduced models

ujjawaljain-msft commented on specification/storage/resource-manager/Microsoft.Storage/stable/2025-08-01/examples/StorageConnectorCRUD/StorageConnectors_Create.json:
```diff
@@ -0,0 +1,86 @@
+          "type": "DataShare",
+          "connection": {
+            "type": "DataShare",
+            "dataShareUri": "azds://eastus:datashare1:12345678-1234-1234-1234-123456789123"
```
Shared the internal document offline with you. `dataShareUri` will be returned in the response of CreateDataShare which the customer has to use while calling CreateConnector API. 

`authProperties.identityResourceId` -> Null/empty if you're using System-Assigned Identity and the ARM resource Id of the identity for User-Assigned Identity. You need to ensure that the identity passed here matches the identity associated with the storage account

ujjawaljain-msft committed:
Introducing new api version 2025-08-01 with examples and readme.md changes

ujjawaljain-msft committed:
Adding StorageConnectors and StorageDataShares resource types to 2025-08-01 version

kah279 committed:
add smart tier support in 2025-08-01

Commit:
Add MockRun trigger type for storage task assignments

- Add MockRun to TriggerType union with documentation
- Include MockRun example for storage task assignment creation
- MockRun trigger type is available in API version 2025-08-01

nicolashou1994 committed:
SRP change for 2025-08-01 version

Add Version Decorator for SRP model change

add @removed(Versions.2025_08_01) for MinimumTlsVersion

Fix the Compile error of SRP on 2025-08-01

change Static Website description

change openApi description

fix format

Address abhishek comment

address abhishek comment

address staticwebsite comment

add real example

address the wrong description

change description

remove extra quote

Add new value to AllowedCopyScope Enum - manusrao

remove minimumTLSVersion change

Fixing CI Validation issues

Fix the StorageAccountPropertiesCreateParameters

format

AbhishekKrishna123 commented on specification/storage/resource-manager/Microsoft.Storage/stable/2025-08-01/openapi.json:
```diff
@@ -11541,23 +11545,13 @@
       "type": "string",
       "description": "Set the minimum TLS version to be permitted on requests to storage. The default interpretation is TLS 1.0 for this property.",
```
Update the description. The default is not TLS 1.0 anymore and it will be TLS 1.2.

AbhishekKrishna123 replied:
We can also add a note that older TLS versions were retired as breaking change in 2025-08-01 API. It would make it easier for people going through the latest documentation to understand vs comparing old and new specs. Might be good to check with ARM guidelines and one more SRP reviewer about this

jackhou1994 replied:
sure, will update the description

AbhishekKrishna123 commented on specification/storage/resource-manager/Microsoft.Storage/stable/2025-08-01/examples/BlobServicesGet.json:
```diff
@@ -89,6 +89,12 @@
+        },
+        "staticWebsite": {
+          "enabled": true,
+          "indexDocument": "default-name-of-index-page-under-each-directory",
```
Please discuss the examples with one more SRP reviewer, and update all the examples

jackhou1994 replied:
okay, i will update as specific example

AbhishekKrishna123 commented on specification/storage/Storage.Management/examples/2025-08-01/StorageAccountCreate.json:
```diff
@@ -55,7 +55,21 @@
     },
     "resourceGroupName": "res9101",
-    "subscriptionId": "00000000-0000-0000-0000-000000000000"
+    "subscriptionId": "00000000-0000-0000-0000-000000000000",
```
As this is an example, use a random GUID for subId

jackhou1994 commented on specification/storage/Storage.Management/models.tsp:
```diff
@@ -1327,6 +1338,32 @@ model DeleteRetentionPolicy {
+  indexDocument?: string;
+
+  /**
+   * The absolute path to a webpage that Azure Storage serves for requests that don't correspond to an existing file.
```
i copied from official doc: 

<img width="934" height="400" alt="Image" src="https://github.com/user-attachments/assets/4ff3c1ba-82f6-4802-ace1-8676844e7b97" />

AbhishekKrishna123 replied:
The official doc is wrong, please change it

AbhishekKrishna123 commented on specification/storage/resource-manager/Microsoft.Storage/stable/2025-08-01/examples/BlobServicesPut.json:
```diff
@@ -76,7 +76,13 @@
+        "isVersioningEnabled": true,
+        "staticWebsite": {
+          "enabled": true,
+          "indexDocument": "default-name-of-index-page-under-each-directory",
```
Thaks for updating the other examples. update this example too

AbhishekKrishna123 commented on specification/storage/Storage.Management/models.tsp:
```diff
@@ -3534,6 +3571,43 @@ model AccountImmutabilityPolicyProperties {
+   */
+  file?: ServiceSharedKeyAccessProperties;
+
+  /**
```
Sure, this is fine.

golddove reviewed (approved):
Approving Storage Task Assignment changes

Commit:
Add stopAssignment endpoint for StorageTaskAssignments

Commit:
Adding Data Collaboration Policy Properties to storage account model for 2025-08-01 verssion

GHeer0295 committed:
Adding tags replication swagger changes

ujjawaljain-msft committed:
1. Add versioning tag to ObjectReplicationPolicyPropertiesTagsReplication
2. Fix StorageAccountCreate.json example for 2025-08-01 version

NishantGoyalMS reviewed (approved):
Approving Data RP related changes.

jackhou1994 commented on specification/storage/resource-manager/Microsoft.Storage/stable/2025-08-01/examples/BlobServicesGet.json:
```diff
@@ -84,7 +84,13 @@
+          "staticWebsite": {
+            "enabled": true,
+            "indexDocument": "home.html",
+            "defaultIndexDocumentPath": "/site/home.html",
```
addressed comment, DefaultIndexDocument is mutually Exclusive with IndexDocument so removed DefaultIndexDocument

blueww replied:
It's better to add this limitation to the parameter description, and make provide this info in server error to make it more clear and actionable.

blueww replied:
I just tried set DefaultIndexDocumentPath without IndexDocument, so still meet the same error "XML specified is not syntactically valid.", fail both with DefaultIndexDocumentPath value as "/site/home.html" or "site/home.html".

```
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
PUT

Absolute Uri:
https://management.azure.com/subscriptions/****/resourceGroups/weitry/providers/Microsoft.Storage/storageAccounts/****/blobServices/default?api-version=2025-08-01

Headers:
x-ms-client-request-id        : 60da216c-17d6-4885-aebe-1109fc684541
accept-language               : en-US

Body:
{
  "properties": {
    "staticWebsite": {
      "enabled": true,
      "defaultIndexDocumentPath": "site/home.html"
    }
  }
}


DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
BadRequest

Headers:
Pragma                        : no-cache
x-ms-request-id               : 8171a5fd-741c-4075-b00a-dc366a52cbdf
Strict-Transport-Security     : max-age=31536000; includeSubDomains
x-ms-operation-identifier     : tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47,objectId=c2bdc6ca-a59c-411c-bf1a-a97ceade71ea/southeastasia/132fae6b-8b92-4930-a79b-91e1ff17c79b
x-ms-ratelimit-remaining-subscription-writes: 799
x-ms-ratelimit-remaining-subscription-global-writes: 11999
x-ms-correlation-request-id   : d538182b-73c1-4f6b-8a30-82609b7621ec
x-ms-routing-request-id       : SOUTHEASTASIA:20260303T031916Z:d538182b-73c1-4f6b-8a30-82609b7621ec
X-Content-Type-Options        : nosniff
X-Cache                       : CONFIG_NOCACHE
X-MSEdge-Ref                  : Ref A: D4315165C9C34C6B96B6A3B800BE7309 Ref B: SG2AA1070305023 Ref C: 2026-03-03T03:19:13Z
Cache-Control                 : no-cache
Date                          : Tue, 03 Mar 2026 03:19:15 GMT

Body:
{
  "error": {
    "code": "ContainerOperationFailure",
    "message": "XML specified is not syntactically valid.\nRequestId:dbbcc212-501e-005e-61bc-aa76ba000000\nTime:2026-03-03T03:19:16.3581045Z"
  }
}
```

blueww replied:
Per the teams discussion, this is an SRP bug for using old FE API version to send request to FE side. 
@jackhou1994 already has a PR to fix it, with the merge ETA as 3/4. (Still need SRP deployment)

nicolashou1994 committed:
Fix Static Website json syntax error

mikeharder committed:
Merge branch 'main' into personal/ujjawaljain/nds-tsp

kah279 commented on specification/storage/resource-manager/Microsoft.Storage/stable/2025-08-01/openapi.json:
```diff
@@ -7642,7 +7648,8 @@
         "Premium",
-        "Cold"
+        "Cold",
+        "Smart"
```
The ZRS and AFEC restrictions are temporary. We will be converging to the behavior in the error message shortly. GA is in a month and we will remove AFEC restriction under this API version before GA. SKU restriction will be lifted after GA but also under this api version. We will communicate both of these in the smart tier preview and GA docs

nicolashou1994 committed:
Fix syntax error in Static Website BlobServicesList.json

kah279 committed:
update access tier description
